### PR TITLE
Introduce MVS and pruning for go mod graph parsing

### PR DESCRIPTION
**Pull request by darscan**
#### What does this PR do?

feat: Introduce MVS and pruning for go mod graph parsing

- Introduces Minimal Version Selection for the go mod graph parser
- Introduces dep-graph pruning (on by default) for go mod graph parser

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+14
```
[the remark this answers is not in the document]
Comment (darscan) [reply .nvmrc:1]: https://github.com/snyk/cli/blob/master/package.json#L24

[the remark this answers is not in the document]
Comment (darscan) [reply .nvmrc:1]: And TSConfig ecmaVersion is now 2019 (which maps to Node 12): https://node.green/#ES2019

File: lib/parsers/gomod-graph-parser.ts [hunk 2]
```diff
@@ -1,63 +1,133 @@
...
+      semver.gt(child.version, maxVersions.get(child.name)!)
+    ) {
+      maxVersions.set(child.name, child.version);
+    }
+  }
+  return maxVersions;
+}
+
+function getChildrenByParent(lines: Line[]) {
+  const childrenByParent: Record<string, { name: string; version: string }[]> =
```
Comment (gitphill) [inline lib/parsers/gomod-graph-parser.ts:119]:
```suggestion
  const childrenByParent: Record<string, Package[]> =
```

File: package.json [hunk 3]
```diff
@@ -30,22 +29,22 @@
   "homepage": "https://github.com/snyk/snyk-go-parser#readme",
   "dependencies": {
     "@iarna/toml": "^2.2.5",
-    "@snyk/dep-graph": "^2.3.0",
+    "@snyk/dep-graph": "^2.6.1",
+    "@snyk/go-semver": "^1.2.1",
```
Comment (gitphill) [inline package.json:33]: do we want to be more specific here?

Comment (darscan) [reply package.json:33]: as in pin?

Comment (gitphill) [reply package.json:33]: maybe, CLI could relock in a break potentially

**Review by JamesPatrickGill — APPROVED** [review]

**Pull request merged**